compiler/ir/backend.jvm/entrypoint/src/org/jetbrains/kotlin/backend/jvm/FacadeClassSourceShimForFragmentCompilation.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.backend.jvm
+
+import org.jetbrains.kotlin.descriptors.SourceFile
+import org.jetbrains.kotlin.fileClasses.JvmFileClassInfo
+import org.jetbrains.kotlin.fileClasses.JvmFileClassUtil.getFileClassInfoNoResolve
+import org.jetbrains.kotlin.load.kotlin.FacadeClassSource
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.resolve.jvm.JvmClassName
+import org.jetbrains.kotlin.resolve.source.PsiSourceFile
+import org.jetbrains.kotlin.serialization.deserialization.IncompatibleVersionErrorData
+import org.jetbrains.kotlin.serialization.deserialization.descriptors.DeserializedContainerAbiStability
+import org.jetbrains.kotlin.serialization.deserialization.descriptors.DeserializedContainerSource
+
+class FacadeClassSourceShimForFragmentCompilation(private val containingFile: PsiSourceFile) :
```

Comment:
How come this class is not used right now?

Reply:
I "see" this (`FacadeClassSourceShim ForFragmentCompilation`) and `FragmentCompilerSymbolTableDecorator` as the only two classes that are used nowhere else but from the plug-in.

I am running the "combined" project set-up, so my IDEA can "see" the uses from the plug-in and does not complain.

No usages planned from the main repo.

Reply:
Yeah, most of the compiler team uses the single project, so these will be marked as unused.

But it's not a big deal. The build on the build server will fail anyway if someone accidentally removes these declarations.